src/screens/AppIntroScreen.tsx

```diff
@@ -0,0 +1,110 @@
+import React, { useEffect } from 'react';
+import { ListRenderItem, ScrollView, StyleSheet, View } from 'react-native';
+import AppIntroSlider from 'react-native-app-intro-slider';
+
+import { BoldText, Image, RegularText, SafeAreaViewFlex, Wrapper } from '../components';
+import { colors, normalize, texts } from '../config';
+import { useStaticContent } from '../hooks';
+import { parseIntroSlides } from '../jsonValidation';
+import { AppIntroSlide } from '../types';
+
+const keyExtractor = (slide: AppIntroSlide) => slide.title + slide.text;
+
+const SliderButton = ({ label }: { label: string }) => {
+  return (
+    <View style={styles.sliderButton}>
+      <BoldText>{label.toUpperCase()}</BoldText>
+    </View>
+  );
+};
+
+type Props = {
+  setOnboardingComplete: () => void;
+};
+
+const renderSlide: ListRenderItem<AppIntroSlide> = ({ item }) => {
+  return (
+    <ScrollView>
+      <Image
+        source={{ uri: item.image }}
+        containerStyle={styles.imageContainer}
+        resizeMode="contain"
+      />
+      <Wrapper style={styles.noPaddingBottom}>
+        <BoldText big center>
+          {item.title.toUpperCase()}
+        </BoldText>
+      </Wrapper>
+      <Wrapper>
+        <RegularText center>{item.text}</RegularText>
+      </Wrapper>
+    </ScrollView>
+  );
+};
+
+export const AppIntroScreen = ({ setOnboardingComplete }: Props) => {
+  const { data: slides, error, loading } = useStaticContent({
+    name: 'appIntroSlides',
+    type: 'json',
+    parseFromJson: parseIntroSlides,
+    fetchPolicy: 'network-only'
+  });
+
+  useEffect(() => {
+    if (error || (!loading && !slides?.length)) {
+      setOnboardingComplete();
+    }
+  }, [error, loading, slides?.length]);
+
+  if (error || loading || !slides?.length) {
+    return null;
+  }
+
+  return (
+    <SafeAreaViewFlex>
+      <AppIntroSlider<AppIntroSlide>
```

Comment:
there is an unexpected behavior when clicking "next" very fast. it somehow jumps around. tested with my older and newer ios. this does not happen with my android! please try out if it happens to you as well.

Reply:
it happens for me as well, and not on android either.

unfortunately we do not provide the whole button component, but only the visual layer to the app intro slider.

what do you think is the best course of action here?

Reply:
all right. please check if you can pass an own button component to avoid "their" problems.
otherwise you can check issues of the repo, if this is public already. otherwise create one and provide a PR with some possible fix?

Reply:
so. it is possible to provide a custom `renderPagination` method, and i tried a bit around.

1. it already is a bit of work to have it look and behave exactly the same (using as much of the already implemented functionality as you can) while creating a custom `onPress` for the next button. (with that i mean having it be its own touchable that uses the `goToSlide` of the `ref` for the app intro slider)
2. after trying around for a while i did not come to a clean solution. everything i tried seems to revolve around the internal state for `activeIndex` not updating fast enough. i tried several workarounds but none solved the problem of the `goToSlide` function being called multiple times with the same values (we need the `activeIndex` to calculate the args for the call).

for now i do not see a way for us to **easily** fix it. neither by using a fork, nor by creating a custom `renderPagination`.

it can probably be done by altering the state management of the `AppIntroSlider` in a more fundamental and complex way, though.

Reply:
https://github.com/Jacse/react-native-app-intro-slider/issues/254

Reply:
let's see when and how they will answer. i also subscribed to the issue. thanks